src/big/big_float.cr

```diff
@@ -15,7 +15,8 @@ struct BigFloat < Float
   def initialize(str : String)
     # Strip leading '+' char to smooth out cases with strings like "+123"
     str = str.lchop('+')
-    LibGMP.mpf_init_set_str(out @mpf, str, 10)
+    err = LibGMP.mpf_init_set_str(out @mpf, str, 10)
```

Comment:
I don't think there is a need for a variable here:
```
if LibGMP.mpf_init_set_str(out @mpf, str, 10) == -1
  raise ArgumentError.new("Invalid BigFloat: #{str}")
end
```
And maybe use `str.dump` or manually put it in quotes.

Reply:
Good points thanks! Fixed.